libs/elasticsearch/langchain_elasticsearch/vectorstores.py

```diff
@@ -13,106 +12,42 @@
     Union,
 )
 
-import numpy as np
 from elasticsearch import Elasticsearch
-from elasticsearch.helpers import BulkIndexError, bulk
+from elasticsearch.helpers.vectorstore import (
+    BM25Strategy,
+    DenseVectorScriptScoreStrategy,
+    DenseVectorStrategy,
+    DistanceMetric,
+    RetrievalStrategy,
+    SparseVectorStrategy,
+)
+from elasticsearch.helpers.vectorstore import (
+    VectorStore as EVectorStore,
+)
 from langchain_core.documents import Document
 from langchain_core.embeddings import Embeddings
 from langchain_core.vectorstores import VectorStore
 
 from langchain_elasticsearch._utilities import (
     DistanceStrategy,
-    maximal_marginal_relevance,
-    model_must_be_deployed,
-    with_user_agent_header,
+    user_agent,
 )
+from langchain_elasticsearch.client import create_elasticsearch_client
+from langchain_elasticsearch.embeddings import EmbeddingServiceAdapter
 
 logger = logging.getLogger(__name__)
 
 
 class BaseRetrievalStrategy(ABC):
```

Comment:
I'm slightly concerned this interface might have imported this / a user's code might rely on this. Should we keep the old implementation and just deprecate it?

Reply:
You are completely right, I got carried away. Keeping the existing ones now with deprecation warning.